samtranslator/model/__init__.py

```diff
@@ -370,10 +370,10 @@ def get_runtime_attr(self, attr_name: str) -> Any:
 
         :return: Dictionary that will resolve to value of the attribute when CloudFormation stack update is executed
         """
+        if attr_name not in self.runtime_attrs:
+            raise KeyError(f"{attr_name} attribute is not supported for resource {self.resource_type}")
```

Comment:
Why is KeyError correct?

Reply:
Spoke offline; KeyError is already used in similar places, so this shouldn't cause confusion and maintains consistency.

Reply:
Put my comment here for visibility: `I definitely didn’t think NotImplemented is the correct exception type if runtime attr is not found. I was thinking either KeyError or RuntimeError. Since other methods in that class thrown KeyError when attributes are not found, I picked it as well.`

Reply:
Open for suggestion for a better alternative.